### PR TITLE
bug 1524846: reduce crawling of edit and login pages

### PR DESCRIPTION
**Pull request by escattone**
I feel little capacity to judge the worth of SEO modifications given the murkiness of that world to me, but this is my understanding of [this understanding](https://github.com/mdn/sprints/issues/939) of [bug 1524846](https://bugzilla.mozilla.org/show_bug.cgi?id=1524846).

### CODE REVIEW

**Comment by jwhitlock** [comment]
Checks on local dev:

* http://localhost:8000/en-US/docs/Learn
   - ✅ Logged out: Edit link is to http://localhost:8000/en-US/users/signin?next=/en-US/docs/Learn%24edit 
   - ✅ Logged in: Edit link is to http://localhost:8000/en-US/docs/Learn$edit 
* http://localhost:8000/fr/docs/Apprendre (existing translation)
   - ✅ Logged out: Modifier link is to http://localhost:8000/fr/users/signin?next=/fr/docs/Apprendre%24edit  
   - ✅ Logged in: Modifier link is to http://localhost:8000/fr/docs/Apprendre$edit  
* http://localhost:8000/en-US/users/signin?next=/en-US/docs/Learn%24edit
   - ✅ Head element is ``<link rel="canonical" href="http://localhost:8000/en-US/users/signin">``
* http://localhost:8000/fr/users/signin?next=/fr/docs/Apprendre%24edit
   - 🤔 Head element is ``<link rel="canonical" href="http://localhost:8000/fr/users/signin">``  

I think a stronger version would be to make the English login page the canonical version.

**Comment by jwhitlock** [comment]
There's a link to ``$edit`` in the ``<menu type="context" id="edit-history-menu">``, added in ``kuma/wiki/jinja2/wiki/document.html``, that could use similar link-swapping behaviour.

**Comment by escattone** [comment]
Thanks for the review @jwhitlock! I think I've addressed the issues you discovered (and went a bit further within `kuma/wiki/jinja2/wiki/document.html`, but I think it's warranted).

**Comment by escattone** [comment]
@jwhitlock Ready for review again. I ran `pytest --base-url http://localhost:8000 tests/headless` with no failures, and `pytest --base-url http://localhost:8000 --driver Chrome --driver-path ~/pytest-selenium/chromedriver tests/functional` with only two failures (both timeouts on signin) so I think we're going to be OK on the functional/headless tests.

File: kuma/wiki/jinja2/wiki/document.html [hunk 1]
```diff
@@ -188,7 +193,7 @@ <h1>{{ document.title }}</h1>
               {% if document.parent_id and document.current_revision %}
                 {% if document.current_revision.localization_in_progress %}
                   {% if not settings.MAINTENANCE_MODE and document.current_revision.translation_age >= 10 %}
-                    <div class="overheadIndicator translationInProgress"><p>{% trans translate_url=document.get_edit_url() %}
+                    <div class="overheadIndicator translationInProgress"><p>{% trans translate_url=doc_edit_url %}
```
Comment (jwhitlock) [inline kuma/wiki/jinja2/wiki/document.html]: Nice catch!

File: kuma/wiki/jinja2/wiki/document.html [hunk 2]
```diff
@@ -301,7 +306,7 @@ <h2 class="offscreen">{{ _('Document Tags and Contributors') }}</h2>
         (function($) {
             $('#wikiArticle').children('h2[id]').each(function() {
                 var id = $(this).attr('id');
-                var href = '{{ document.get_edit_url() }}#' + id;
+                var href = '{{ doc_edit_url }}#' + id;
```
Comment (jwhitlock) [inline kuma/wiki/jinja2/wiki/document.html]: again, nice catch! I'm not sure section editing will survive going through login, but I think we only show it for some logged in users. Maybe no users? My memory is failing me.

**Pull request merged**